D.gitignore

```diff
@@ -0,0 +1,36 @@
+# Gitignore for projects in D using dub.
+# Dlang: http://dlang.org/
+# Dub:   https://github.com/D-Programming-Language/dub
+
+## Binary files ##
+
+# Object files
+*.o
+*.obj
+
+# Libraries
+*.a
+*.lib
+*.so
+*.dylib
+*.dll
+
+# Executables
+*.exe
+
+
+## Dub specific ##
+.dub/
+
+# Result of 'dub test'
+__test__*__
+
+# Documentation artifacts (dub --build=[docs|ddox])
+docs.json
+__dummy.html
+docs/*
+
+# By default, ignore dub.selections.json.
+# You should usually rely on tags in dub.json rather than pinning your project
+# to a certain commit, but if you really need to, remove this line:
+dub.selections.json
```

Comment:
It should be by default. That's why I added a lengthy comment to explain that it can safely be removed, but it's addition to source control should be discouraged, because it defeats the purpose of have semantic versions.